src/client/proxy/http.cc

```diff
@@ -390,41 +423,98 @@ bool HTTPMessage::ExtractIncomingRequest() {
   return true;
 }
 
-void HTTPMessage::HandleJumpService() {
-  // TODO(GUZZI) should have boolean return value; error
-  // response?; research this
+bool HTTPMessage::HandleJumpService()
+{
   // TODO(anonimal): add support for remaining services /
   // rewrite this function
-  std::size_t pos1 = m_Path.rfind(m_JumpService.at(0));
-  std::size_t pos2 = m_Path.rfind(m_JumpService.at(1));
-  std::size_t pos;
-  if (pos1 == std::string::npos) {
-    if (pos2 == std::string::npos)
-      return;  // Not a jump service
-    else
-      pos = pos2;
-  } else {
-    if (pos2 == std::string::npos)
-      pos = pos1;
-    else if (pos1 > pos2)
-      pos = pos1;
-    else
-      pos = pos2;
-  }
-  auto base64 = m_Path.substr(pos + m_JumpService.at(0).size());
-  // We must decode
-  base64 = boost::network::uri::decoded(base64);
-  // Insert into address book
-  LOG(debug)
-    << "HTTPProxyHandler: jump service for " << m_Address
-    << " found at " << base64 << ", inserting to address book";
-  // TODO(unassigned): this is very dangerous and broken.
-  // We should ask the user for confirmation before proceeding.
-  // Previous reference: http://pastethis.i2p/raw/pn5fL4YNJL7OSWj3Sc6N/
-  // We *could* redirect the user again to avoid dirtiness in the browser
-  kovri::client::context.GetAddressBook().InsertAddressIntoStorage(
-      m_Address, base64);
-  m_Path.erase(pos);
+
+  // Perform sanity check again to:
+  // - ensure valid jump service request
+  // - update pos with m_JumpService sub-string index (hacky)
```

Comment:
Using a variable name is bad documentation style because A) it doesn't accurate describe the function in plain language (which is the point of documentation) and B) the name will need to be changed in any future refactoring (if the refactorer remembers to do so).